### ISSUE TITLE
Suggest droppable positions while dragging

### ISSUE THREAD

**Issue by jai1331** [react-dnd/react-dnd#1575]
Thanks for the awesome library but I'm stuck at this requirement, how do we show position or placeholder to drop items while drag an item like below:

![Capture](https://user-images.githubusercontent.com/8530380/66542944-93704e80-eb51-11e9-982e-134f3e89aa1f.PNG)

So here, the left side is a simple list which contains list items.

Left side is :

1. This is not droppable, draggable only
2. After drag, this list remains there same all the time.

On the right side, the dragged item will be dropped here.

1. Once the user dragged an item from left side, the other side/ right side should indicate a placeholders to user where they can drop.(placeholder in the sense droppable area letting users know you can drop it here).
2. All the dropped items on the right side can be swappable within right side.

I'm not sure how do I achieve this, since I'm using this awesome library for small to-do and list drag and drop functionalities.

**Comment by mmissey** [comment]
Are you using the new api? Either way, you can use the `canDrop` property to style the drop areas:
```
const [dropProps, drop] = useDrop({
    accept: values(ModuleDragDropTypes),
    canDrop: (item, monitor) => dropTarget.canDrop(props, monitor),
    drop: (item, monitor) => dropTarget.drop(props, monitor),
    collect: monitor => ({
      isOver: !!monitor.isOver(),
      canDrop: !!monitor.canDrop(),
      item: monitor.getItem(),
    }),
  });

  // Show dropzones when moving (wired into canDrop prop from React DnD)
  // Also show dropzones when adding new modules
  <DropArea className=`${dropProps.canDrop ? 'droppable' : ''}` />
```

**Comment by jai1331** [comment]
@mmissey will try this out and let you know shortly. Thanks for the suggestion. Just a query, by using canDrop am I able to show droppable areas while dragging an item to drop ?

**Comment by vladbicu** [comment]
@jai1331, response for your query: yes. You can highlight droppable areas while dragging.

**Comment by jai1331** [comment]
@vladbicu Thanks, but i need something similar to this https://react-email-editor-demo.netlify.com/. When a user dragging an item from one side to canvas/droppable area. I need to show drop-zones/suggestible droppable areas to drop those items. 

After dragged the first item, Now the user will drag a next item to the droppable area where I need to show, they can drop that item like top/bottom/left/right of the first dropped item. You can refer more on this https://stackoverflow.com/questions/58183717/react-drag-and-drop-plugin-suggestion

**Comment by Kivylius** [comment]
**FIY** this can be **closed** as the the answer was provided over at stack-overflow:
https://stackoverflow.com/questions/58183717/react-drag-and-drop-plugin-suggestion

**Comment by darthtrevino** [comment]
This may be worth turning into an official documentation example

**Comment by jai1331** [comment]
@darthtrevino just for my clarification, Is it possible to show/create drop areas while hovering some item on the right side like an example above. When dragging an item from left side to right side, consider there's an item already placed, now while dragging/hovering that left side item on right side item, I should able to show drop area dynamically to drop that item.

**Comment by darthtrevino** [comment]
I don't see why not - a drop target can detect when it's being hovered over and emit some signal about a placeholder needing to be shown

**Comment by jai1331** [comment]
You can check this simple example https://stackblitz.com/edit/react-7fwk4k?file=Container.js, where user will drag list 1 from left side to drop area, after dropping list 1 then the user will drag list 2 when the list 2 hover above on/mouseover on list 1 , I should show drop areas showing like you can drop these areas only while hover particular list item not every time. Hope this can helps to understand better.

**Comment by mmissey** [comment]
Like this? https://stackblitz.com/edit/react-vsfxa9

**Comment by jai1331** [comment]
@mmissey  Thanks, but it shows drop area only when you move drag item to bottom or top of List 1. What I'm trying to achieve is when you hover on/above on List 1 while dragging second item List 2, it should show list of drop areas like top, bottom, left, right to drop that List 2 item. Hope this helps. Requirement is : When you mouseover on already dropped item, it should show possible dropareas.

**Comment by jai1331** [comment]
@darthtrevino @mmissey any possibilities on this ?

**Comment by jdebr** [comment]
@jai1331 I think what you want is to apply styling to the drop areas as a function of 1. drag is in progress and 2. mouse is over a dropped list item.  You can use the useDragLayer hook to inject isDragging into the Container component from mmissey's example, then you need to implement detection for when you are over your dropped items and expose that state to the Container element (react dnd won't really help you with this, as your are just interacting with your own List components).  You can then use the combination of these two states to toggle the appropriate styling on the DropAreas.

I was trying to throw together a quick example but I accidentally refreshed and lost my work and now I'm out of time.  It gets a little tricky when detecting if the pointer is over your list item while dragging because the drag preview blocks pointer events - you can try to add styling 'pointer-events:none' when drag starts to help with this.

**Comment by jai1331** [comment]
@jdebr Thanks , it's my bad luck, will do a workaround on this and it would be great if I have an example to get an idea. Thanks to you man will sort this out and let you know,.

**Comment by saifmattias** [comment]
@jai1331 Two years later, did you sort it out? I am trying to archive the same thing, Any example you can share?

**Comment by jai1331** [comment]
@saifmattias yes you can create custom droppable position check [here]( https://stackblitz.com/edit/react-dzipd7) If you have doubts let me know, I am happy to help you out. Use `useDrop()` to create dynamic drop.
